pkg/resources/managed_account_test.go

```diff
@@ -39,6 +40,19 @@ func TestManagedAccountCreate(t *testing.T) {
 	})
 }
 
+func TestManagedAccountRead(t *testing.T) {
+	r := require.New(t)
+	d := managedAccount(t, "test-account", map[string]interface{}{"name": "test-account"})
+	q := snowflake.ManagedAccount(d.Id()).Show()
+
+	WithMockDb(t, func(db *sql.DB, mock sqlmock.Sqlmock) {
+		// Test when resource is not found
+		mock.ExpectQuery(q).WillReturnError(sql.ErrNoRows)
+		err := resources.ReadManagedAccount(d, db)
```

Comment:
I think we should find some way to assert that after these Read functions are called that the object has been removed from the state.